docs/classic-ui/theming/diazo.md

```diff
@@ -0,0 +1,178 @@
+{term}`Diazo` allows you to apply a theme contained in a static HTML web page to a dynamic website created using any server-side technology.
+With Diazo, you can take an HTML wireframe created by a web designer, and turn it into a theme for your favourite CMS, redesign the user interface of a legacy web application without even having access to the original source code, or build a unified user experience across multiple disparate systems, all in a matter of hours, not weeks.
+
+A Diazo theme consists of a static HTML page, referred to as the _theme_, and a rules file, conventionally called {file}`rules.xml`.
+
+When using Diazo, you will work with syntax and concepts familiar from working with HTML and CSS.
+And by allowing you to seamlessly integrate XSLT into your rules files, Diazo makes common cases simple, and complex requirements possible.
+
+
+## Create an add-on package
+
+To create a Diazo theme, you need to create an add-on package with [`plonecli`](https://github.com/plone/plonecli).
+
+You can run `plonecli` using [pipx](https://pipx.pypa.io/stable/), without needing to install `plonecli`, as follows.
+
+```shell
+pipx run plonecli create addon diazo.theme
+```
+
+Answer all the questions.
+
+Next, add a theme called `theme` to the add-on package using `plonecli`.
+
+```shell
+cd diazo.theme
+pipx run plonecli add theme
+```
+
+Answer the question of the theme name.
+
+
+### Theme structure
+
+The previous step creates a {file}`theme` folder inside {file}`diazo.theme/src/diazo/theme`, with the following structure.
+
+```console
+.
+├── index.html
+├── manifest.cfg
+├── package.json
+├── README.rst
+├── rules.xml
+├── styles
+│   ├── theme.min.css
+│   └── theme.scss
+└── tinymce-templates
+    ├── bs-dark-hero.html
+    ├── bs-hero-left.html
+    └── bs-pricing.html
+```
+
+The purpose of each item is described as follows.
+
+{file}`index.html`
+:   This is the theme file.
+    Plone populates it with its content, according to rules defined in {file}`rules.xml`.
+
+{file}`manifest.cfg`
+:   This file contains the theme configuration, such as the theme name, the path to the rules file and some other configurations.
+
+{file}`package.json`
+:   The theme folder is, by itself, a JavaScript package.
+    If you want to develop your theme here, you can declare the dependencies of your theme in this file.
+    For instance, you can add `bootstrap` or `tailwind` as dependencies, and manage those dependencies and the building of your final CSS from here.
+
+{file}`README.rst`
+:   The file that explains how this theme is built and developed.
+
+{file}`rules.xml`
+:   The file where you will write your rules to bring Plone content into the theme in the {file}`index.html` file.
+
+{file}`styles` folder
+:   Where you can save your theme's CSS files.
+
+{file}`tinymce-templates`
+:   Template files that can be loaded into the TinyMCE editor in Plone.
+    It requires additional configuration in the add-on's profile's {file}`registry.xml` file.
```

Comment:
I searched within the Classic UI folder for a reference to the file `registry.xml`, but came up empty. Is that the correct filename? There is documentation on TinyMCE in https://6.docs.plone.org/classic-ui/tinymce-customization.html that uses the filename `profiles/default/registry/tinymce.xml`.

Where can I get more information? We should add a `seealso` admonition to whatever that reference should be, in case someone wants to fiddle with TinyMCE. Here's my guess. Please let me know. Thank you!

````suggestion
    It requires additional configuration in the add-on's profile's {file}`registry.xml` file.

    ```{seealso}
    [TinyMCE customization](https://6.docs.plone.org/classic-ui/tinymce-customization.html)
    ```
````

Reply:
The file `registry.xml` present in `profiles/default` can be split in several files under `profiles/default/registry` with arbitrary names ending in `.xml`, so `registry.xml` and `profiles/default/registry/tinymce.xml` is valid and `profiles/default/registry/plone.xml` is valid.

The docs lack information about tinymce templates. I will add a PR with that information and then we can link there.

Reply:
@stevepiercy I have added the relevant information in the tinymce docs in this very PR. 

I tried to use the same example content as created by the plonecli template. 

Let's see if it is understandable :)

Reply:
Yup, this is great! I tidied it up a bit. However, you should take a look at a compatibility issue with Plone 6.1/TinyMCE 7 and the Template plugin in https://plone6--1870.org.readthedocs.build/classic-ui/tinymce-customization.html#insert-preconfigured-html-blocks. 😞 

I also revised the suggestion, and added your comment as a tip to the TinyMCE docs.

Please let me know. Thank you!

Reply:
Good question. There was a [similar issue with the accordion plugin](https://6.docs.plone.org/backend/upgrading/version-specific-migration/upgrade-to-61.html#tinymce-upgraded-in-classic-ui). It appears that there was work on the Template plugin in mockup in https://github.com/plone/mockup/pull/1384. @petschki can you advise?

I don't know if this requires an entry in the 6.1 upgrade guide.

Reply:
the `template` plugin has moved to a premium plugin since tiny version `6.4` (see https://www.tiny.cloud/docs/tinymce/latest/advanced-templates/) but we've reimplemented the "community plugin" for version 7.x in mockup. So its there and it works in Plone 6.1 too.